Added condition for link without title

rohan-goel opened the pull request:
## Fixes
<!-- If PR doesn't fully resolve the issue, replace 'Fixes' below with 'Related to'. -->
<!-- If there is no issue being resolved, open one before creating this pull request. -->
Fixes #297 by @obulat 

## Description
Added condition to add link to **This work** when no title is provided and removed **property="dct:title"** .
<!-- Concisely describe what the pull request does. -->

## Checklist
<!-- Replace  the [ ] with [x] to check the boxes. --> 
- [x] My pull request has a descriptive title (not a vague title like `Update
  index.md`).
- [x] My pull request targets the *default* branch of the repository (`main` or `master`).
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [ ] I added tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no
  visible errors.

[best_practices]:https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53

## Developer Certificate of Origin
<!-- You must read and understand the following attestation. -->

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

rohan-goel commented:
I hard coded "This work". How can I change it dynamically?

obulat commented on src/utils/license-utilities.js:
```diff
@@ -184,7 +184,7 @@ function generateCreatorCode(creatorName, creatorProfileUrl) {
 /**
  * Generate html for work title:
- * 1. If the work title is blank, event if work link is provided, return blank string
+ * 1. If the work title is blank, even if work url is provided, return blank string
```
We will have to add another parameter, something like `isTitleDefault`, to distinguish between default title(which will not have `dct:title` added to it, and the title provided by the user.

obulat replied:
I guess I couldn't clearly explain what I meant.

Chooser has i18n library which does internationalization. We have access to its $t function in components, but not in the `license-utilities.js`. 

We do the translation in the component (`LicenseHTML` in this case), where we have access to i18n's $t function. However, when we pass the translated string to `license-utilities.js`, we have no way of guessing if the string we pass as a title is provided by the user (something like "Cats playing outside") or the default "This work" that has been translated (so, for example, "Эта работа" in Russian). If we simply compare with "This work", we miss the translated default titles and treat them as user-provided titles.

To solve this problem, we can call `generateHTML` with a title parameter that is equal to ` this.attributionDetails.workTitle || this.$t('license-use.richtext.workTitle')` and a third Boolean parameter, which is true if the user has provided a title, or false if we are using a default title.

Then, in the `license-utilities.js` we have the following branches:
1. if title is default and we have a link to work: we add an '<a>' anchor with relevant properties, but don't use 'dct:title'.
2. if title is NOT default and we have a link: do the same but add 'dct:title'
3. other branches should not change, I guess.

obulat commented on src/utils/license-utilities.js:
```diff
@@ -196,10 +196,15 @@ function generateWorkCode(workTitle, workUrl) {
     if (workTitle) {
```
Here, the condition will be simplified, because `workTitle` will never be blank

obulat commented:
> I hard coded "This work". How can I change it dynamically?

This issue turned out to be more complex than I expected.
I think the best way will be to send the title (user-provided, or the translated string for 'This work') as a parameter to this function here:
https://github.com/creativecommons/chooser/blob/404fea77f103cf36fe54e8a3607eaf5ad0e8e1cf/src/components/LicenseHTML.vue#L24

We can add the translated 'This work', or the actual user-provided title to `this.attributionDetails`, something like:
`const attributionDetails = {...this.attributionDetails, workTitle: work || this.$t('license-use.richtext.workTitle')}`, 
and then call the generateHTML function.

https://github.com/creativecommons/chooser/blob/404fea77f103cf36fe54e8a3607eaf5ad0e8e1cf/src/components/LicenseHTML.vue#L26

As I commented inline, we will also need to add another parameter for showing if the title is default or not.

rohan-goel commented:
I have added a parameter called **defaultTitle** , this will never be empty and added **showProperty** to toggle **property="dct:title"**  if title is default or user provided.

obulat reviewed (changes requested):
Please see the comments inline, @rohan-goel

rohan-goel commented:
I added isTitleDefault parameter to add and remove property="dct:title" accordingly. Now property will be added only if user provided title is present.

obulat reviewed (approved):
Thank you for your contribution, @rohan-goel!
I noticed that the code needed some complex refactoring to make it simpler, so I added that, and made the tests more comprehensive.
